docs/user/spark.md

```diff
@@ -37,17 +38,19 @@ see [Build DAOS Hadoop Filesystem](#builddaos) for details.
 
 ### JAR Files
 
-`daos-java-<version>.jar` and `hadoop-daos-<version>.jar` need to be deployed
-on every compute node that runs Spark or Hadoop.
-Place them in a directory, e.g., `$SPARK_HOME/jars` for Spark and
-`$HADOOP_HOME/share/hadoop/common/lib` for Hadoop, which is accessible to all
-the nodes or copy them to every node.<br/>
+`daos-java-<version>-protobuf3-netty4-shaded.jar` and
+`hadoop-daos-<version>-protobuf3-netty4-shaded.jar` need to be deployed on
+every compute node that runs Spark or Hadoop. Place them in a directory,
+e.g., `$SPARK_HOME/jars` for Spark and `$HADOOP_HOME/share/hadoop/common/lib`
+for Hadoop, which is accessible to all the nodes or copy them to every node.
+<br/>
 
-### `daos-site-example.xml`
+### `core-site-daos-ref.xml` (version >= 2.2.1, or `daos-site-example.xml`)
 
-Extract from `hadoop-daos-<version>.jar` and rename to `daos-site.xml`. Then
-copy it to your application config directory, e.g., `$SPARK_HOME/conf` for
-Spark and `$HADOOP_HOME/etc/hadoop` for Hadoop.
+Extract from `hadoop-daos-<version>-protobuf3-netty4-shaded.jar`. Then merge
+with your Hadoop `core-site.xml` which is under your application config
+directory, e.g., `$SPARK_HOME/conf` for Spark and `$HADOOP_HOME/etc/hadoop`
```

Comment:
Is there supposed to be a core-site.xml file under `$SPARK_HOME/conf`. My Spark download originally did not have any `core-site.xml` which I can merge with `core-site-daos-ref.xml`. I do see it for Hadoop under `$HADOOP_HOME/etc/hadoop` button for Spark.

Reply:
Good catch.

Spark can work with or without Hadoop installation. If it's with Hadoop inst., you only need to merge DAOS related properties into $HADOOP_HOME/etc/hadoop/core-site.xml. If it's without Hadoop inst., you have two ways to put your core-site.xml. One is to put it in $SPARK_HOME/conf. The other is to put the directory to which your HADOOP_CONF_DIR env points.

I need to rephrase it. thanks.